File: backend/app/models/certification.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

# Request Models
class CertificationCreate(BaseModel):
    title: str
    start_date: str
    end_date: str  # Can be "Present" or date
    instructor: Optional[str] = None
    platform: str
    certification_link: Optional[str] = None

class CertificationUpdate(BaseModel):
    title: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    instructor: Optional[str] = None
    set_instructor: Optional[bool] = None
    platform: Optional[str] = None
    certification_link: Optional[str] = None
    set_certification_link: Optional[bool] = None

# Response Models
class CertificationResponse(BaseModel):
    id: str
    user_id: str
    title: str
    start_date: str
    end_date: str
    instructor: Optional[str] = None
    platform: str
    certification_link: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/api/certification.ts
import api from './axios'

export interface CertificationResponse {
  id: number
  user_id: number
  title: string
  start_date: string
  end_date: string
  instructor: string | null
  platform: string
  certification_link: string | null
  created_at: string
  updated_at: string
}

export interface CertificationCreate {
  title: string
  start_date: string
  end_date: string
  instructor?: string | null
  platform: string
  certification_link?: string | null
}

export interface CertificationUpdate {
  title?: string
  start_date?: string
  end_date?: string
  instructor?: string | null
  platform?: string
  certification_link?: string | null
  set_instructor?: boolean
  set_certification_link?: boolean
}

export const certificationApi = {
  getCertification: async (): Promise<CertificationResponse[]> => {
    const response = await api.get<CertificationResponse[]>('/api/certification/')
    return response.data
  },

  getCertificationById: async (certificationId: number): Promise<CertificationResponse> => {
    const response = await api.get<CertificationResponse>(`/api/certification/${certificationId}`)
    return response.data
  },

  createCertification: async (data: CertificationCreate): Promise<CertificationResponse> => {
    const response = await api.post<CertificationResponse>('/api/certification/', data)
    return response.data
  },

  updateCertification: async (certificationId: number, data: CertificationUpdate): Promise<CertificationResponse> => {
    const response = await api.put<CertificationResponse>(`/api/certification/${certificationId}`, data)
    return response.data
  },

  deleteCertification: async (certificationId: number): Promise<void> => {
    await api.delete(`/api/certification/${certificationId}`)
  },
}
